apps-rendering/src/components/Callout/calloutForm.tsx

```diff
@@ -0,0 +1,186 @@
+import type { FormField as FormFieldType } from '@guardian/apps-rendering-api-models/formField';
+import type { ArticleFormat } from '@guardian/libs';
+import {
+	Button,
+	InlineError,
+	InlineSuccess,
+	Link,
+} from '@guardian/source-react-components';
+import { useState } from 'react';
+import type { FC } from 'react';
+import { ContactText, Disclaimer, FormField } from './formFields';
+import { ShareLink } from './shareLink';
+import { calloutForm, calloutSubmitButton } from './styles';
+
+export interface CalloutFormProps {
+	format: ArticleFormat;
+	id: number;
+	fields: FormFieldType[];
+	disableInputs?: boolean;
+}
+
+type SubmitDataType = Record<string, string | string[]>;
+export type FormDataType = { [key in string]: undefined | string | string[] };
+export type ValidationErrors = { [key in string]: string };
+
+const CALLOUT_URL =
+	'https://callouts.code.dev-guardianapis.com/formstack-campaign/submit';
+
+const CalloutForm: FC<CalloutFormProps> = ({ id, fields, format }) => {
+	const [formData, setFormData] = useState<FormDataType>({});
+	const [validationErrors, setValidationErrors] = useState<ValidationErrors>(
+		{},
+	);
+	const [submissionError, setSubmissionError] = useState<string>('');
+	const [submissionSuccess, setSubmissionSuccess] = useState<boolean>(false);
+
+	const setFieldInFormData = (
+		id: string,
+		data: string | string[] | undefined,
+	): void => {
+		const currentErrors = validationErrors;
+		currentErrors[id] = '';
+		setValidationErrors(currentErrors);
+
+		setFormData({
+			...formData,
+			[id]: data,
+		});
+	};
+
+	const validateForm = (): boolean => {
+		const errors: ValidationErrors = {};
+		let isValid = true;
+		fields.forEach((field: FormFieldType) => {
+			if (field.mandatory && !formData[field.id]) {
+				errors[field.id] = 'This field is required';
+				isValid = false;
+			}
+			if (field.type === 'email' && formData[field.id]) {
+				const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+				if (!emailRegex.test(formData[field.id] as string)) {
+					errors[field.id] = 'Please enter a valid email address';
+					isValid = false;
+				}
+			}
+			if (
+				['number', 'phone'].includes(field.type) &&
+				formData[field.id]
+			) {
+				const numberRegex = /^[0-9]+$/;
+				if (!numberRegex.test(formData[field.id] as string)) {
+					errors[field.id] = 'Please enter a valid number';
+					isValid = false;
+				}
+			}
+			return isValid;
+		});
+		setValidationErrors(errors);
+		return Object.keys(errors).length === 0;
+	};
+
+	const onSubmit = async (formData: FormDataType): Promise<void> => {
+		// Reset error for new submission attempt
+		setSubmissionError('');
+		const isValid = validateForm();
+		if (!isValid) return;
+
+		// need to add prefix `field_` to all keys in form
```

Comment:
more for curiosity but how come we need this prefix?

Reply:
Its a good question, its required by the [formstack api ](https://developers.formstack.com/reference/form-id-submission-post). I'll update the comment here to be more informative.